src/api/Filter.js

```diff
@@ -0,0 +1,48 @@
+import { addExtremeValuesToArr } from '../utils/addExtremeValuesToArr';
+import { FilterTypes } from './FilterQueryBuilder';
+
+const filterFunctions = {
+  [FilterTypes.IN](filterValues, featureValue) {
+    return filterValues.includes(featureValue);
+  },
+  [FilterTypes.BETWEEN](filterValues, featureValue) {
+    const checkRange = (range) => {
+      const [lowerBound, upperBound] = range;
+      return featureValue >= lowerBound && featureValue <= upperBound;
+    };
+
+    return addExtremeValuesToArr(filterValues).some(checkRange);
+  }
+};
+
+export function applyFilter({ filters = {}, returnedType = 'boolean' }) {
+  if (!Object.keys(filters).length) {
+    return () => returnedType === 'number' ? 1 : true;
+  }
+
+  return object => {
+    const columns = Object.keys(filters);
+
+    const featurePassesFilter = columns.every(column => {
```

Comment:
object is too generic, better use `feature`, don't you think?

Reply:
`object` is more related with deck.gl..